src/shared/test/integration/rpc-bridge-test.js

```diff
@@ -0,0 +1,185 @@
+import Bridge from '../../bridge';
+
+describe('RPC-Bridge integration', () => {
+  const sandbox = sinon.createSandbox();
+  let clock;
+  let port1;
+  let port2;
+  let bridges = [];
+
+  function createBridge() {
+    const bridge = new Bridge();
+    bridges.push(bridge);
+    return bridge;
+  }
+
+  function waitForMessageDelivery() {
+    return new Promise(resolve => setTimeout(resolve, 0));
+  }
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+    const channel = new MessageChannel();
+    port1 = channel.port1;
+    port2 = channel.port2;
+  });
+
+  afterEach(() => {
+    bridges.forEach(bridge => bridge.destroy());
+    sandbox.restore();
+    clock.restore();
+  });
+
+  describe('establishing a connection', () => {
+    it('should invoke Bridge `onConnect` callbacks after connecting', async () => {
+      clock.restore();
+      const bridge = createBridge();
+      const reciprocalBridge = createBridge();
+
+      let callbackCount = 0;
+      const callback = () => {
+        ++callbackCount;
+      };
+
+      bridge.onConnect(callback);
+      bridge.onConnect(callback); // allows multiple callbacks to be registered
+      reciprocalBridge.onConnect(callback);
+
+      const channel = bridge.createChannel(port1);
+      reciprocalBridge.createChannel(port2);
+
+      await waitForMessageDelivery();
+      assert.equal(callbackCount, 3);
+
+      // Additional calls to the RPC `connect` method are ignored
+      channel.call('connect');
+      reciprocalBridge.call('connect');
+
+      await waitForMessageDelivery();
+      assert.equal(callbackCount, 3);
+    });
+  });
+
+  describe('sending and receiving RPC messages', () => {
+    it('should invoke Bridge method handler on every channel when calling a RPC method', async () => {
+      const bridge = createBridge();
+      const otherChannel = new MessageChannel();
+      bridge.createChannel(port1);
+      bridge.createChannel(otherChannel.port1);
+      const reciprocalBridge1 = createBridge();
+      const reciprocalBridge2 = createBridge();
+      reciprocalBridge1.on('method1', (arg, cb) => cb(null, `${arg}foo`));
+      reciprocalBridge2.on('method1', (arg, cb) => cb(null, `${arg}bar`));
+      reciprocalBridge1.createChannel(port2);
+      reciprocalBridge2.createChannel(otherChannel.port2);
+
+      const results = await bridge.call('method1', 'param1');
```

Comment:
```suggestion
      const results = await bridge.call('method1', 'param1');
      
```

Having blank lines before and after the "action" phase of a test with many lines makes it easier to see which code is the setup, which code is the action being tested and which code is the assertions.